### ISSUE TITLE
[no-empty-interface] Should allow single extends when specifying a generic

### ISSUE THREAD

**Issue by SLKnutson** [typescript-eslint/typescript-eslint#4119]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-empty-interface": ["error"]
  }
}
```

```TS
import { Immutable } from 'immer';

interface TestInterfaceMutable {
  a: string;
}

interface TestInterface extends Immutable<TestInterfaceMutable> {
}
```

**Expected Result**

No errors

**Actual Result**

`  7:11  error  An interface declaring no members is equivalent to its supertype  @typescript-eslint/no-empty-interface`

**Additional Info**

Since the result is different than the interface it extends from, I would consider this to be a very similar situation to the other exception to the rule, where unions are allowed (`InterfaceA & InterfaceB`)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.33` |
| `@typescript-eslint/parser`        | `^4.33.0` |
| `TypeScript`                       | `~4.3.5` |
| `ESLint`                           | `^7.32.0` |
| `node`                             | `X.Y.Z` |
| `immer`                           | `^9.0.6`|

**Thanks!!!**

**Comment by bradzacher** [comment]
```ts
type TestInterface = Immutable<TestInterfaceMutable>
```

Is what you should use instead

**Comment by SLKnutson** [comment]
@bradzacher 
Using a type is a good work around, but I have a strong preference for using an interface in the scenario when the underlying interface has many fields, or the underlying fields have complex types. In this scenario, typescript error message usability suffers greatly. For instance, consider this code where a field that doesn't exist is assigned a value:
```
import { Immutable } from 'immer';

interface TestInterfaceMutable {
  a: string;
  b: number;
  c: boolean;
  d?: string;
  e?: number;
  f?: boolean;
  g?: HTMLDivElement;
  h?: bigint;
  readonly i?: number;
}

interface TestInterface extends Immutable<TestInterfaceMutable> {
}

const usingInterface: TestInterface = {};
usingInterface.j = 'hi';

type TestType = Immutable<TestInterfaceMutable>;

const usingType: TestType = {};
usingType.j = 'hi';
```
The typescript error for assigning a non existent field for the interface is concise: `TS2339: Property 'j' does not exist on type 'TestInterface'.`
However, the typescript error for assigning a non existent field for the type is not as helpful: `TS2339: Property 'j' does not exist on type '{ readonly a: string; readonly b: number; readonly c: boolean; readonly d?: string | undefined; readonly e?: number | undefined; readonly f?: boolean | undefined; readonly g?: { readonly align: string; ... 244 more ...; readonly focus: (options?: FocusOptions | undefined) => void; } | undefined; readonly h?: bigint ...'.`

This isn't typescript-eslint's fault, but I think it should be taken into consideration. The existing exception allowing unions of types could also be done with a type instead of an interface, but special accommodation for it was made in the rule.

Thanks!

**Comment by bradzacher** [comment]
It's worth mentioning that you're free to use the `allowSingleExtends` option.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-interface.md#options
